### DISCUSSION TITLE
Show number of links and backlinks to notes in finders

### DISCUSSION THREAD

**Discussion by lambtho12** [nvim-telekasten/telekasten.nvim#36] [Ideas]
In order to improve and cleanup my zettels, I would like to quickly see which notes are poorly linked so I can improve them by creating new links.

It would be great to have a finder that lists the notes without many links in it or links to it. As for the tag finder, it could show the number of links directly in the finder. In fact, the finder could show all notes and the number of links in/to it, ordering them from lowest count to highest.

**Comment by renerocksai** [comment]
Looking for orphaned notes (no or n links to them) and finding notes with at least n links in them is part of my special search buffer thing. So it is def on my list. Might take a bit, though.

Currently, the tag finder only searches for tags. So it knows which notes contain which tags. It is easy to count them there.

OK, now I think I see what you mean. When you search for a tag, you see a list of notes that contain the tag. In the file list, you want to see the number of links from and to each note. And a way to sort by either of them. 

With a bit of work, that is doable. It will slow things down, probably. But we can try. Extra key mappings could switch between sort modes. 

Because of potential performance implications, I would make this feature optional, with a config option.

It could go into all or most finders, I guess. So I would rename that feature to something along the lines of 'show number of links and backlinks to notes in finders'.

I have seen multi-column tabular pickers in telescope - so maybe that's an option, too, for displaying the notes.

**Reply by renerocksai** [reply]
I was so free to update the title. Didn't know GH lets you do this :smile:

**Reply by Liquidmantis** [reply]
This is high on my wishlist, as well, but unfortunately it's likely beyond my meager skills to attempt to help on.  Like we were discussing in the other thread this support would remove my primary dependency on Obsidian.

**Reply by lambtho12** [reply]
> OK, now I think I see what you mean. When you search for a tag, you see a list of notes that contain the tag. In the file list, you want to see the number of links from and to each note. And a way to sort by either of them.

More or less yes. But I do not necessarily want that on the normal picker (find_files). When I use find_file(), I know what I am doing and what file I am looking for. 

What I propose here is a separate picker that would show and list files with poor linking specifically. That would prevent unnecessary complicating (and slowing) the default find_files().

The workflow where I would use such picker would be something like
1. I create a new zettel
2. I look for tags or related files using find_files, find_tag, and the other existing pickers
3. I make a final check to see if there are some poor linked files that I could link to the new zettel in order to improve them while I am at it (-> This is where the specific picker would come handy)

As @Liquidmantis says, this is something that is trivial with Obsidian (I never used Obsidian my guess is that many people actually use the graph to find poorly linked notes in an instant).

**Reply by Liquidmantis** [reply]
I've got a string-along request!  Finding files that mention the note title or a tag but aren't actually linked.  

This is the Obsidian feature:
<img width="303" alt="image" src="https://user-images.githubusercontent.com/399056/145688652-ffb66c2a-ddd6-4407-ad8a-a97c370e245b.png">

This just happened to be while I had a dummy test file named 'bar' open in Obsidian.

**Comment by renerocksai** [comment]
With 89191a1c8457e609649f780cf71138370e614a49, the tag search following the tag selection now shows the number of links from and to a note.

Show all tags, filter down to my desired one:

![image](https://user-images.githubusercontent.com/30892199/145767714-9442aa28-0b29-4c32-85ed-729609b75b72.png)

Select it (`#blog`) aaaaaaaaand:

![image](https://user-images.githubusercontent.com/30892199/145767897-a6be5985-48e3-4cd7-8627-945007c36d14.png)

L ... number of links in this note
B ... number of backlinks to this note

It is not polished yet, there is no special sorting, and you can't configure bounds specifying what you'd consider a low link count, etc.

But if you don't have hundreds of matches, this can actually be useful.

**Comment by renerocksai** [comment]
Just fixed the highlights of the link counts and devicons 

![image](https://user-images.githubusercontent.com/30892199/146629201-141f7e39-e05c-4b4a-9e2f-f8b677b9e638.png)
